File: jar/src/main/java/jline/api/measures/Ms_matusita.java
package jline.api.measures;

import jline.util.matrix.Matrix;

public final class Ms_matusita {
    private Ms_matusita() {}

    /**
     * Matusita distance between two probability distributions.
     * Part of the fidelity family.
     *
     * @param P first probability distribution
     * @param Q second probability distribution
     * @return Matusita distance
     */
    public static double ms_matusita(Matrix P, Matrix Q) {
        if (P.length() != Q.length()) {
            throw new IllegalArgumentException("Distributions must have the same size");
        }

        double sum = 0.0;
        for (int i = 0; i < P.length(); i++) {
            double diff = Math.sqrt(P.get(i)) - Math.sqrt(Q.get(i));
            sum += diff * diff;
        }

        return Math.sqrt(sum);
    }
}

File: jar/src/main/kotlin/jline/api/measures/Ms_matusita.kt
package jline.api.measures

import jline.util.matrix.Matrix
import kotlin.math.sqrt

/**
 * Matusita distance between two probability distributions.
 * Part of the fidelity family.
 * 
 * @param P first probability distribution
 * @param Q second probability distribution
 * @return Matusita distance
 */
fun ms_matusita(P: Matrix, Q: Matrix): Double {
    require(P.length() == Q.length()) { "Distributions must have the same size" }
    
    var sum = 0.0
    for (i in 0 until P.length()) {
        val diff = sqrt(P.get(i)) - sqrt(Q.get(i))
        sum += diff * diff
    }
    
    return sqrt(sum)
}
/**
 * Matusita metric algorithms
 */
@Suppress("unused")
class MsMatusitaAlgo {
    companion object {
        // Class documentation marker for Dokka
    }
}